### PR TITLE
Fix/myforms menu update

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Summary | Résumé

Completes the menu links to be functional and localized. The main change is adding a "Copy to clipboard" feature to published cards. The menu links are still a work-in-progress and point to the most relevant part of form-builder currently available.

# Test instructions | Instructions pour tester la modification

Open a published card menu. There should be a "Copy to clipboard" button. Clicking it should result in a success message and the card URL copied to your clipboard.

Open a draft card menu and instead of a "Copy to clipboard" button, there should be an "Edit link. Clicking that link or any other menu link should all navigate to the form-builder (only the clipboard button does not navigate).

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

Menu links should be updated if/once the form-build links are refactored.

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

**Review by craigzour — CHANGES_REQUESTED** [review]

File: components/myforms/Card/Card.tsx [hunk 1]
```diff
@@ -59,20 +59,22 @@ export const Card = (props: CardProps): React.ReactElement => {
             title={t("card.menu.more")}
             items={[
               {
-                title: "Edit",
-                url: `/form-builder/create/${id}`,
+                title: isPublished ? t("card.menu.copyLink") : t("card.menu.edit"),
+                url: isPublished
+                  ? `/${i18n.language}/id/${id}/settings`
+                  : `/form-builder/create/${id}`,
```
[the remark this answers is not in the document]
Comment (thiessenp-cds) [reply components/myforms/Card/Card.tsx:65]: I'm trying to remember why I had that down as "create'. If I remember right that was the best route at the time but I agree it doesn't make sense

[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply components/myforms/Card/Card.tsx:65]: the `/form-builder/edit` route doesn't exist... only `create` exists so far.

[the remark this answers is not in the document]
Comment (craigzour) [reply components/myforms/Card/Card.tsx:65]: I do not get it. I was pretty sure it was the path I used to edit an existing form when I was working on https://github.com/cds-snc/platform-forms-client/pull/1265 . Can't remember what I was doing exactly but I was able to load a form in the form builder

[the remark this answers is not in the document]
Comment (thiessenp-cds) [reply components/myforms/Card/Card.tsx]: Sure, done. (will update the Edit link when a more complete edit path exists)

**Comment by timarney** [comment]
> Copy to clipboard currently uses an Alert() to show success. (future work to either update or remove)

FYI - Just created and merge a Dialog component that you could repurpose a bit 

See screenshots here: https://github.com/cds-snc/platform-forms-client/pull/1316#issue-1460358911

https://github.com/cds-snc/platform-forms-client/pull/1316/files#diff-addd4fbf791c7d09fecf1775f0ef6b6177dd2d81c5a080836f9c0ae821d6e8b1

**Comment by thiessenp-cds** [comment]
> 

Thanks! That would certainly be better than an Alert(). I'll ask Sam about it

**Comment by thiessenp-cds** [comment]
> > Copy to clipboard currently uses an Alert() to show success. (future work to either update or remove)
> 
> RE: Alert
> 
> I just created and merged a Dialog component that you could repurpose a bit
> 
> See screenshots here: [#1316 (comment)](https://github.com/cds-snc/platform-forms-client/pull/1316#issue-1460358911)
> 
> https://github.com/cds-snc/platform-forms-client/pull/1316/files#diff-addd4fbf791c7d09fecf1775f0ef6b6177dd2d81c5a080836f9c0ae821d6e8b1

We went with an inline text updated but I'm sure that dialog will be re-used in the future :)

File: components/myforms/Card/Card.tsx [hunk 2]
```diff
@@ -59,20 +59,22 @@ export const Card = (props: CardProps): React.ReactElement => {
             title={t("card.menu.more")}
             items={[
               {
-                title: "Edit",
-                url: `/form-builder/create/${id}`,
+                title: isPublished ? t("card.menu.copyLink") : t("card.menu.edit"),
+                url: isPublished
+                  ? `/${i18n.language}/id/${id}/settings`
```
[the remark this answers is not in the document]
Comment (thiessenp-cds) [reply components/myforms/Card/Card.tsx]: I updated the links but some link are sort of a work in progress. I'll mark this as resolved for now but please bring it up again if the links still look wrong

**Review by craigzour — DISMISSED** [review]
LGTM :)

**Comment by thiessenp-cds** [comment]
> LGTM :)

I seem to have lost a library in between starting and finishing writing this code :D It umm is back now

**Review by craigzour — APPROVED** [review]

**Pull request merged**